src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java

```diff
@@ -153,4 +153,26 @@ private static JavaSparkContext createTestSparkContext(Map<String, String> overr
         final SparkConf sparkConf = setupSparkConf("TestContext", DEFAULT_SPARK_MASTER, DEFAULT_TEST_PROPERTIES, overridingProperties);
         return new JavaSparkContext(sparkConf);
     }
+
+    /**
+     * Determine the number of cores Spark master should use. Only used in Spark Test
+     *   Read the specification from the environmental variable GATK_TEST_SPARK_CORES
+     *      If the value is a valid positive integer, use it
+     *      If the value is bogus (strings, etc), or the env. var. is not set, use all available cores, as in "local[*]"
+     */
+
+    private static String determineSparkMaster() {
+        int foo = 0;
+        try {
+            foo = Integer.parseInt( System.getenv("GATK_TEST_SPARK_CORES") );
+        } catch ( NumberFormatException e ) {}
```

Comment:
If `GATK_TEST_SPARK_CORES` is set (ie., `System.getenv()` returned non-null for it) and is non-numeric, throw a `UserException` with a descriptive error message instead of silently swallowing the error.